extension/background/intentRunner.js

```diff
@@ -531,6 +531,42 @@ export function getRegisteredNicknames() {
   return registeredNicknames;
 }
 
+const registeredPageName = {};
+
+export function registerPageName(name, context) {
```

Comment:
Instead of context this should take `{url, creationDate}` (or maybe fill in creationDate itself)